zerver/views/submessage.py

```diff
@@ -27,11 +29,14 @@ def process_submessage(request: HttpRequest,
     except orjson.JSONDecodeError:
         return json_error(_("Invalid json for submessage"))
 
-    do_add_submessage(
-        realm=user_profile.realm,
-        sender_id=user_profile.id,
-        message_id=message.id,
-        msg_type=msg_type,
-        content=content,
-    )
+    try:
+        do_add_submessage(
+            realm=user_profile.realm,
+            sender_id=user_profile.id,
+            message_id=message.id,
+            msg_type=msg_type,
+            content=content,
+        )
+    except IntegrityError:
+        raise JsonableError(_("Parent message was deleted."))
```

Comment:
I'm wondering here whether we should be doing the `IntegrityError` catch or use `access_message` with `lock_message=True`. What was your reasoning for choosing to catch?

Reply:
Message row doesn't get updated in this codepath, so I thought it would be better performance wise to not lock it.

Reply:
Yeah, that's what's making me wonder too. Perhaps let's leave it as you've done it and wait for Tim or other reviewers.

Reply:
I think it's correct to lock the message here -- if we think of the SubMessage as effectively a model containing extra fields for the Message (just like Reaction), `lock_message=True`  seems like the right semantics.  Certainly any change to SubMessage is going to end up flushing the Message from all caches, etc.